source/DG.HostApp/Services/ClusterConfigActualizer.cs

```diff
@@ -0,0 +1,120 @@
+                                this.applicationOrchestrator.UnRegister(instance.Key);
+                                continue;
+                            }
+
+                            // case when instance was moved to another host
+                            if (clusterInstanceDescription.PlacementPolicies.First(pp => pp == currentHost.Name) == null)
+                            {
+                                this.applicationOrchestrator.Stop(instance.Key);
+                                this.applicationOrchestrator.UnRegister(instance.Key);
+                                continue;
+                            }
+
+                            // case when amount of instance was changed
+                            int nodeInstanceAmountDiference = this.InstanceDifferenceCalculator(clusterInstanceDescription, currentHost, instance.Value.Count);
+
+                            if (nodeInstanceAmountDiference > 0)
+                            {
+                                this.applicationOrchestrator.BuildInstance(clusterInstanceDescription.Type, clusterInstanceDescription.Name, clusterInstanceDescription.Settings?.ToString(), nodeInstanceAmountDiference);
+                            }
+                            else if (nodeInstanceAmountDiference < 0)
+                            {
+                                this.applicationOrchestrator.RemoveInstance(clusterInstanceDescription.Type, clusterInstanceDescription.Name, Math.Abs(nodeInstanceAmountDiference));
+                            }
+                        }
+
+                        // starting the required applications
+                        foreach (var application in applicationInstances)
+                        {
+                            var isInstanceExist = instances.ContainsKey(ApplicationExtensions.ConstructUniqueId(application.Type, application.Name));
+                            var isHostedonThisNode = application.PlacementPolicies.First(x => x == currentHost.Name).Any();
+
+                            if (isHostedonThisNode && !isInstanceExist)
+                            {
+                                int nodeInstanceAmount = this.InstanceDifferenceCalculator(application, currentHost, 0);
+
+                                this.applicationOrchestrator.BuildInstance(application.Type, application.Name, application.Settings?.ToString(), nodeInstanceAmount);
+                            }
+                        }
+
+                        Thread.Sleep(5000);
+                    }
+                }, cancellationToken);
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            return Task.Run(() => { Console.WriteLine("ServiceWatcher stopped"); }, cancellationToken);
+        }
+
+        private int InstanceDifferenceCalculator(ApplicationInstance applicationInstance, Core.Model.ClusterConfig.Host currnetHost, int existingInstanceAmount)
```

Comment:
Can we move this logic to separate class? we can write unit tests to cover this calculation logic and use direct call for this function.